### PR TITLE
1558: Move the category color generation code into Java/Wicket

### PR DESCRIPTION
**Pull request by payten**
... from the custom Javascript and also store this on the GradebookUISettings session object to retain the color state for the duration of the wicket session

### CODE REVIEW

**Comment by payten** [comment]
Delivers #1558.

File: gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/model/GradebookUiSettings.java [hunk 1]
```diff
@@ -78,4 +80,12 @@ public void setCategoryScoreVisibility(final String category, final Boolean visi
 	public String toString() {
 		return ToStringBuilder.reflectionToString(this, ToStringStyle.MULTI_LINE_STYLE);
 	}
+
+	public void setCategoryColor(String category, String color) {
+		categoryColors.put(category, color);
+	}
+
+	public String getCategoryColor(String category) {
```
Comment (steveswinsburg) [inline gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/model/GradebookUiSettings.java]: Is this name or id? Sorry to be a pain, but could it be made more explicit as to which one, or just add doco?

File: gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/ToggleGradeItemsToolbarPanel.java [hunk 2]
```diff
@@ -79,7 +80,20 @@ protected void populateItem(final ListItem<String> categoryItem) {
...
 
-				categoryItem.add(new Label("category", categoryName));
+				GradebookUiSettings settings = gradebookPage.getUiSettings();
+				if (settings == null) {
+					settings = new GradebookUiSettings();
+					gradebookPage.setUiSettings(settings);
+				}
+
+				if (settings.getCategoryColor(categoryName) == null) {
+					settings.setCategoryColor(categoryName, generateRandomRGBColorString());
```
Comment (steveswinsburg) [inline gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/panels/ToggleGradeItemsToolbarPanel.java]: Ok it's name, yeah just update the above so the parameter is categoryName :+1:

**Comment by payten** [comment]
Thanks @steveswinsburg! I've cleaned up those parameter names in `GradebookUiSettings.java`.

**Pull request merged**